.github/workflows/ci.yml

```diff
@@ -0,0 +1,85 @@
+name: CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+
+jobs:
+  tox:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        tox-env:
+          - py38
+          - black-template
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install dependencies
+        run: |
+          python -m pip install -U pip
+          python -m pip install -U tox
+      - name: Tox ${{ matrix.tox-env }}
+        run: tox -e ${{ matrix.tox-env }}
+
+  docker:
```

Comment:
why not just directly translate from travis by using matrix for docker and bare metal?

Reply:
Right now, I could probably merge the `docker` and `tox` jobs, but I'm hoping that we could speed up each job leter by adding some specialised caching. 

The bare metal job requires some extra services that the other jobs don't use.